### PR TITLE
add tcp_flags

### PR DESCRIPTION
**Pull request by allewwaly**
add tcp_flags OXM to support floodlight matching with tcp_flags
as discussed in https://groups.google.com/a/openflowhub.org/forum/#!topic/floodlight-dev/uSnA0IgkBJs

### CODE REVIEW

**Comment by bsn-abat** [comment]
Can one of the admins verify this patch?

**Comment by bsn-abat** [comment]
Can one of the admins verify this patch?

**Comment by rizard** [comment]
@andi-bigswitch this is blocking a couple Floodlight OSS pull requests. It's the addition of an OXM and LGTM. Mind if I merge?

**Comment by andi-bigswitch** [comment]
@rizard LGTM, but I didn't look at the except bit definitions of the flags. Provided you checked those LGTM.

**Comment by allewwaly** [comment]
From http://openvswitch.org/support/dist-docs/ovs-ofctl.8.txt
in the  tcp_flags options:

> tcp_flags=[+flag...][-flag...]
>               Bitwise  match on TCP flags.  The flags and mask are 16-bit num‐
>               bers written in decimal or in hexadecimal prefixed by 0x.   Each
>               1-bit  in mask requires that the corresponding bit in flags must
>               match.  Each 0-bit in mask causes the corresponding  bit  to  be
>               ignored.
> 
>               Alternatively,  the  flags  can  be  specified by their symbolic
>               names (listed below), each preceded by either + for a flag  that
>               must  be  set,  or  - for a flag that must be unset, without any
>               other delimiters between the flags.   Flags  not  mentioned  are
>               wildcarded.   For  example,  tcp,tcp_flags=+syn-ack  matches TCP
>               SYNs that are not ACKs.
> 
>               TCP protocol currently defines 9 flag  bits,  and  additional  3
>               bits  are  reserved (must be transmitted as zero), see RFCs 793,
>               3168, and 3540.  The flag bits are,  numbering  from  the  least
>               significant bit:
> 
>               0: fin No more data from sender.
> 
>               1: syn Synchronize sequence numbers.
> 
>               2: rst Reset the connection.
> 
>               3: psh Push function.
> 
>               4: ack Acknowledgement field significant.
> 
>               5: urg Urgent pointer field significant.
> 
>               6: ece ECN Echo.
> 
>               7: cwr Congestion Windows Reduced.
> 
>               8: ns  Nonce Sum.
> 
>               9-11:  Reserved.
> 
>               12-15: Not matchable, must be zero.

**Comment by rizard** [comment]
Thanks @allewwaly! This LGTM.

ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2016.09.05.1253-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: AUTO-REJECT:  Your change failed pre-merge validation defined in loxigen/.abat-automerge. See log for details.


INSTRUCTIONS: Please fix your problem by

1) Reviewing the logs below and `git commit`ing a fix.

2) Use `git push` to add new commits to this same remote branch/pull request.

3) Get another code review approval from your co-worker indicated by a line of "ACK" in comment, or add another comment yourself with a line of "R_E_T_R_Y (Remove the underscores)". That will cause abat to try to merge your code again.



Log file at http://vss1/abat/2016.09.05.1253-m.loxigen.master/abat.log
/cc

File: java_gen/java_type.py [hunk 1]
```diff
@@ -684,6 +684,8 @@ def gen_fixed_length_string_jtype(length):
 
         'of_oxm_tcp_flags' : { 'value' : u16obj },
         'of_oxm_tcp_flags_masked' : { 'value' : u16obj, 'value_mask' : u16obj },
+        'of_oxm_ovs_tcp_flags' : { 'value' : u16obj },
+        'of_oxm_ovs_tcp_flags_masked' : { 'value' : u16obj, 'value_mask' : u16obj }
```
Comment (rizard) [inline java_gen/java_type.py]: @allewwaly I think you're missing a comma at the end of this line

Comment (allewwaly) [inline java_gen/java_type.py]: Sorry for that. should I update it and re-pull? Or can you just modify it directly?

Comment (rizard) [inline java_gen/java_type.py]: Please update with a commit to this pull request

**Comment by rizard** [comment]
Reviewer: trivial

RETRY

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2016.09.05.1405-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2016.09.05.1405-m.loxigen.master/abat.log
/cc

**Pull request merged**